pkg/virt-controller/watch/network_test.go

```diff
@@ -167,36 +169,89 @@ var _ = Describe("Network interface hot{un}plug", func() {
 	)
 
 	DescribeTable("VM status interfaces requests",
-		func(ifaces []v1.Interface, ifaceRequests, expectedIfaceRequests []v1.VirtualMachineInterfaceRequest) {
-			vm := kubecli.NewMinimalVM("test")
-			vm.Spec.Template = &v1.VirtualMachineInstanceTemplateSpec{}
-			vm.Spec.Template.Spec.Domain.Devices.Interfaces = ifaces
+		func(vmIfaces, vmiIfaces []v1.Interface, ifaceRequests, expectedIfaceRequests []v1.VirtualMachineInterfaceRequest) {
+			vmi := api.NewMinimalVMI("test")
+			vmi.Spec.Domain.Devices.Interfaces = vmiIfaces
+			vm := VirtualMachineFromVMI("test", vmi, true)
+			vm.Spec.Template.Spec.Domain.Devices.Interfaces = vmIfaces
 			vm.Status.InterfaceRequests = ifaceRequests
 
-			trimDoneInterfaceRequests(vm)
+			trimDoneInterfaceRequests(vm, vmi)
 
 			Expect(vm.Status.InterfaceRequests).To(Equal(expectedIfaceRequests))
 		},
 		Entry("have request removed on successful hotplug",
+			[]v1.Interface{{Name: "blue"}},
 			[]v1.Interface{{Name: "blue"}},
 			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
 			[]v1.VirtualMachineInterfaceRequest{},
 		),
 		Entry("keep interface request for pending hotplug",
+			[]v1.Interface{},
+			[]v1.Interface{},
+			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+		),
+		Entry("keep interface request for pending hotplug, when VMI spec has not been updated yet",
+			[]v1.Interface{{Name: "blue"}},
 			[]v1.Interface{},
 			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
 			[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
 		),
+		Entry("keep interface request for pending hotplug, when VM template spec has not been updated yet",
```

Comment:
Here we enter the gray area where we really do not know what to do.
All actions are expected to pass thought the VM first, so to me this hints that the removal is expected from the next restart.

The system is expected to reconcile based on the VM spec, so it sounds like an undefined state to see something like this unless it was an intentional user edit of the VM.

Reply:
Yes, and that is wrong.